crates/conjure_core/src/rule_engine/rule.rs

```diff
@@ -63,7 +63,11 @@ pub struct Reduction {
 pub type ApplicationResult = Result<Reduction, ApplicationError>;
 
 impl Reduction {
-    pub fn new(new_expression: Expression, new_top: Expression, symbols: SymbolTable) -> Self {
+    pub fn new(
+        new_expression: Vec<Expression>,
```

Comment:
@YehorBoiar this shouldn't be a vec. it's the replacement expression, so you want to replace expression with expression. this is probably the reason for the error you are getting.